pkg/k8s/cep/cep.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+// Package cep contains the CiliumEndpoint event handling logic from Kubernetes.
+package cep
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/k8s/types"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/source"
+	"github.com/cilium/cilium/pkg/u8proto"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "k8s-watcher-cep")
+
+func (s *Subscriber) OnAddCiliumEndpoint(ep *types.CiliumEndpoint) error {
+	return s.OnUpdateCiliumEndpoint(nil, ep)
+}
+
+func (s *Subscriber) OnUpdateCiliumEndpoint(oldCEP, newCEP *types.CiliumEndpoint) error {
```

Comment:
This logic is almost pure copy-paste from k8s watcher, for which the main motivation is to clean up the k8s watcher and slowly remove business logic from it.